frontend/src/js/components/PageViewer/Page.tsx

```diff
@@ -46,7 +48,14 @@ export const Page: FC<PageProps> = ({
           const node = containerRef.current;
           if (node) {
             setScale(preview.scale);
-            node.appendChild(preview.canvas);
+
+            const oldCanvas = node.getElementsByTagName('canvas');
```

Comment:
Since you're only expecting one element, you could maybe do `.querySelector` which will return `null` or the element

Reply:
great suggestion 👍